File: RaftSwift/Sources/RaftCore/Utils/Types/RaftConfig.swift
/// Configuration for the Raft node.
public struct RaftConfig: Sendable {
    /// The range of election timeout in milliseconds
    public let electionTimeoutRange: ClosedRange<Int> = 500 ... 1000

    /// The interval of heartbeats in milliseconds
    public let heartbeatInterval = 50

    /// The threshold of log entries to compact
    public let compactionThreshold = 1000

    /// Creates a new configuration
    public init() {}
}

File: RaftKotlin/raft-node/src/main/kotlin/raft/core/utils/types/RaftConfig.kt
package raft.core.utils.types

/**
 * Configuration for the Raft node.
 */
data class RaftConfig(
    /**
     * The range of election timeout in milliseconds
     */
    val electionTimeoutRange: IntRange = 500 .. 1000,
    /**
     * The interval of heartbeats in milliseconds
     */
    val heartbeatInterval: Int = 50,
)